backends/redis.go

```diff
@@ -55,99 +55,84 @@ func (e KeyError) Error() string {
 	return "Key not found"
 }
 
-func (r *RedisBackend) newRedisClusterPool() redis.UniversalClient {
+func (r *RedisBackend) Connect() error {
 	redisLogger.Info("Creating new Redis connection pool")
 
-	timeout := 5 * time.Second
-
-	if r.config.Timeout > 0 {
-		timeout = time.Duration(r.config.Timeout) * time.Second
-	}
-
-	var tlsConfig *tls.Config
-	if r.config.UseSSL {
-		tlsConfig = &tls.Config{
-			InsecureSkipVerify: r.config.SSLInsecureSkipVerify,
-		}
-	}
-
-	var client redis.UniversalClient
-	opts := &redis.UniversalOptions{
-		MasterName:       r.config.MasterName,
-		SentinelPassword: r.config.SentinelPassword,
-		Addrs:            r.getRedisAddrs(),
-		DB:               r.config.Database,
-		Username:         r.config.Username,
-		Password:         r.config.Password,
-		IdleTimeout:      240 * time.Second,
-		ReadTimeout:      timeout,
-		WriteTimeout:     timeout,
-		DialTimeout:      timeout,
-		TLSConfig:        tlsConfig,
+	conf := r.config
+	optsR := model.RedisOptions{
+		Username:         conf.Username,
+		Password:         conf.Password,
+		Host:             conf.Host,
+		Port:             conf.Port,
+		Timeout:          conf.Timeout,
+		Hosts:            conf.Hosts,
+		Addrs:            conf.Addrs,
+		MasterName:       conf.MasterName,
+		SentinelPassword: conf.SentinelPassword,
+		Database:         conf.Database,
+		MaxActive:        conf.MaxActive,
+		EnableCluster:    conf.EnableCluster,
+	}
+
+	tls := model.TLS{
+		Enable:             conf.UseSSL,
```

Comment:
Is UseSSL behaving the same as before?

Reply:
yes, any concern that am not aware of? so far for SSL we only had these 2 configs: useSSL and SSLInsecureSkipVerify, we added support for new configs